File: app/src/main/java/com/ihadzhi/eatlimination/data/Diet.java
package com.ihadzhi.eatlimination.data;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.util.Date;

@Entity
public class Diet {

    public void setId(long id) {
        this.id = id;
    }

    @NonNull
    @PrimaryKey(autoGenerate = true)
    private long id;

    private boolean active;

    private Date createdAt;

    private Date startedOn;

    public Diet(boolean active, Date createdAt, Date startedOn) {
        this.active = active;
        this.createdAt = createdAt;
        this.startedOn = startedOn;
    }

    @Ignore
    public Diet(boolean active, Date startedOn) {
        this(active, new Date(), startedOn);
    }

    @NonNull
    public long getId() {
        return id;
    }

    public boolean isActive() {
        return active;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public Date getStartedOn() {
        return startedOn;
    }
}

File: app/src/main/java/com/ihadzhi/eatlimination/data/Diet.kt
package com.ihadzhi.eatlimination.data

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import java.util.*

@Entity
class Diet(val isActive: Boolean, val createdAt: Date, val startedOn: Date?) {

    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    @Ignore
    constructor(active: Boolean, startedOn: Date?) : this(active, Date(), startedOn)

}